Makefile

```diff
@@ -338,11 +340,17 @@ endif
 # PPA includes the required mongodb-server binaries.
 install-snap-dependencies:
 ## install-snap-dependencies: Install the supported snap dependencies
-ifeq ($(shell go version | grep -o "go1.18" || true),go1.18)
-	@echo Using installed go-1.18
+ifeq ($(shell if [ "$(GO_INSTALLED_VERSION)" \> "$(GO_MOD_VERSION)" -o "$(GO_INSTALLED_VERSION)" = "$(GO_MOD_VERSION)" ]; then echo 1; fi),1)
```

Comment:
This is a bit hacky, is there really no nicer way to do this? Like a plain `if` statement, or even `ifleq`?

Reply:
I do not think there exists ifleq/iflt/etc for Makefiles so comparing greater/less than in Makefiles will require some shell call. So that rules out ifleq or the like

Also  this Makefile doesn't specify SHELL. That means it uses /bin/sh instead of /bin/bash . That means it requires the use of  the  sh inequality syntax instead of bash which means you also lose the bash flexibility for "less than" comparisons vs sh .  So that rules out a nicer plain if. 

So for a cleaner Makefile you _could_ change the Makefile from /bin/sh to /bin/bash - but it would mean testing the older "shell" commands to make sure they are sh -> bash compatible. 

I tried 

`ifdef GO_INSTALLED_VERSION
`
but since that variable is set by parsing "go version" that variable is defined, even if go isn't installed. So you have to compare to the empty string and not use ifdef or ifndef. If there was a GO variable set by the env we could test that, but there does not seem to be, so that's out. 

I removed "cut" so the Makefile is now truly not dependent on coreutils (nice catch!) and repushed. 

If you want a more compact implementation for these conditionals I one could specify /bin/bash for the shell instead of /bin/sh . While that would replace "[ A< B -o A = B ]"  with  [[ A <= B ]] .... it would also require testing the 20 or so shell commands already existing in the Makefile and that's not something I would recommend unless you plan that as part of a larger testing routine.

Reply:
No worries, this is fine. Let's not change too much.